src/main/java/rs/teslaris/core/service/impl/OrganisationUnitServiceImpl.java

```diff
@@ -1,22 +1,259 @@
+import rs.teslaris.core.service.DocumentFileService;
+import rs.teslaris.core.service.MultilingualContentService;
 import rs.teslaris.core.service.OrganisationUnitService;
+import rs.teslaris.core.service.ResearchAreaService;
 
 @Service
 @RequiredArgsConstructor
+@Transactional
 public class OrganisationUnitServiceImpl implements OrganisationUnitService {
 
-    private final OrganisationalUnitRepository organisationalUnitRepository;
+    private final OrganisationUnitRepository organisationUnitRepository;
+
+    private final MultilingualContentService multilingualContentService;
+
+    private final OrganisationUnitsRelationRepository organisationUnitsRelationRepository;
+
+    private final ResearchAreaService researchAreaService;
+    private final DocumentFileService documentFileService;
+
+    @Value("${relation.approved_by_default}")
+    private Boolean relationApprovedByDefault;
+
+    @Value("${organisation_unit.approved_by_default}")
+    private Boolean organisationUnitApprovedByDefault;
+
+    @Override
+    public OrganisationUnit findOrganisationUnitById(Integer id) {
+        return organisationUnitRepository.findByIdWithLangDataAndResearchArea(id).orElseThrow(
+            () -> new NotFoundException("Organisation unit with given ID does not exist."));
+    }
+
+    @Override
+    @Transactional
+    public Page<OrganisationUnitDTO> findOrganisationUnits(Pageable pageable) {
+        return organisationUnitRepository.findAllWithLangData(pageable).map(OrganisationUnitConverter::toDTO);
+    }
+
+    @Override
+    public OrganisationUnitsRelation findOrganisationUnitsRelationById(Integer id) {
+        return organisationUnitsRelationRepository.findById(id).orElseThrow(
+            () -> new NotFoundException(
+                "Organisation units relation with given ID does not exist."));
+    }
+
+    @Override
+    public Page<OrganisationUnitsRelationResponseDTO> getOrganisationUnitsRelations(
+        Integer sourceId, Integer targetId, Pageable pageable) {
+        return organisationUnitsRelationRepository.getRelationsForOrganisationUnits(pageable,
+            sourceId, targetId).map(RelationToRelationDTO::toResponseDTO);
+    }
+
+    @Override
+    public OrganisationUnit getReferenceToOrganisationUnitById(Integer id) {
+        return id == null ? null : organisationUnitRepository.getReferenceById(id);
+    }
 
     @Override
-    public OrganisationUnit findOrganisationalUnitById(Integer id) {
-        return organisationalUnitRepository.findById(id)
-            .orElseThrow(() -> new NotFoundException(
-                "Organisational unit with given ID does not exist."));
+    public OrganisationUnit createOrganisationalUnit(
+        OrganisationUnitDTORequest organisationUnitDTORequest) {
+        OrganisationUnit organisationUnit = new OrganisationUnit();
+        organisationUnit.setName(
+            multilingualContentService.getMultilingualContent(organisationUnitDTORequest.getName())
```

Comment:
Maybe we should consider creating a `setCommon` function that would set all fields and it could be called both from create and from update methods.

Reply:
lets do this later if it can make our code better organized